base/dates/parse.jl

```diff
@@ -281,9 +281,14 @@ function Base.tryparse{T<:TimeType}(
     pos, len = start(str), endof(str)
     values, pos = tryparsenext_internal(T, str, pos, len, df, false)
     if isnull(values)
-        Nullable{T}()
+        return Nullable{T}()
     else
-        Nullable{T}(T(unsafe_get(values)...))
+        try
+            parsed = Nullable{T}(T(unsafe_get(values)...))
```

Comment:
The only other way I can think to do this is to specialize on each subtype of `TimeType` and duplicate the checks done by the type constructors. That's kind of intractable.